File: python/p1.py


"""
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
"""

def sum_three_and_five(x):
    return sum(filter(lambda x: x % 3 == 0 or x % 5 == 0, xrange(3, x)))


if __name__ == '__main__':
    print sum_three_and_five(1000)

File: js/p1.js
var divisible = function(x){
    if (x % 3 === 0 || x % 5 == 0){
        return true;
    }else { return false; }
};

var sum = function(ary){
    var total = 0;
    for (x in ary){
        total += ary[x]
    }
    return total;
};

var multiples = function(n){
    var values = [];
    for(var i = 0; i < n ; i++){
        if (divisible(i)){
            values.push(i)
        }
    }
    return sum(values);
};


// use underscore library below //

var uu= require('underscore')

console.log(multiples(1000))


